src/main/java/net/sf/mpxj/phoenix/Phoenix5Reader.java

```diff
@@ -451,15 +487,28 @@ private void processActivities(Storepoint phoenixProject)
     */
    private void processActivity(Activity activity)
    {
-      Task task = getParentTask(activity).addTask();
+      Task task;
+
+      if (m_useActivityCodesForTaskHierarchy)
+      {
+         task = getParentTask(activity).addTask();
+      }
+      else
+      {
+         task = m_projectFile.addTask();
+
+         // Activity codes
+         populateActivityCodes(task, getActivityCodes(activity));
+      }
+
       task.setActivityID(activity.getId());
 
       task.setActualDuration(activity.getActualDuration());
       task.setActualFinish(activity.getActualFinish());
       task.setActualStart(activity.getActualStart());
       //activity.getBaseunit()
       //activity.getBilled()
-      //activity.getCalendar()
+      task.setCalendar(m_projectFile.getCalendarByName(activity.getCalendar()));
```

Comment:
The updates to set the calendar seem unrelated to the activity code changes, can we create a separate PR for these? It's easier for me to see the changes when I run my test suite with the modified code.

Reply:
Yup. We will split this into 2 PRs.